### DISCUSSION TITLE
Pesan tidak terkirim

### DISCUSSION THREAD

**Discussion by satrijo** [kirimwa-id/forum#31] [Tanya Jawab] [answered]
Saya sudah mencoba beberapa hari menggunakan kirimwa.id yg saya rasakan adalah pesan kadang terkirim kadang tidak, beberapa kali saya coba untuk merestart ataupun membiarkan layar dan aplikasi wa terbuka siapa tau bisa mengatasi tapi ternyata nihil, apakah ada cara agar pesan yang dikirim tidak gagal?

**Comment by rioastamal** [comment, marked as answer]
@satrijo Pesan errornya seperti apa? Bisa dilihat dari response message menggunakan API `GET /messages/MESSAGE_ID` atau melihat semua pesan terkirim secara keseluruhan dengan `GET /messages?status=fail`. Referensi https://developer.kirimwa.id/api/v1/?shell#get-messages

**Reply by satrijo** [reply]
keterangan errornya, Error: Send message timeout after 10000 ms. 
kira2 kenapa ya mas?

**Reply by rioastamal** [reply]
Itu berarti API KirimWA.id tidak bisa menghubungi WhatsApp pada device setelah menunggu 10 detik. Hal itu terjadi biasanya karena masalah koneksi internet di perangkat. Bisa karena internet diperangkat terputus atau sedang lambat. Jika masih mendapatkan timeout terus coba untuk scan ulang QR code dengan memanggi API `GET /qr`

**Reply by rioastamal** [reply]
Bisa juga dicoba settings ini agar WhatsApp tetap aktif meskipun HP dalam keadaan sleep. https://faq.whatsapp.com/android/troubleshooting/notification-problems/?lang=en

**Reply by satrijo** [reply]
Hallo mas, setelah saya coba ganti device ternyata memang sepertinya ada masalah di koneksi device satunya. Semoga tidak ada problem yg lain.
